### PR TITLE
Try to make backend_gdk compatible with numpy 1.6

### LINKED ISSUE #5078: numpy incompatibility
We have non np 1.6 compatible code in the gdk backend which we do not test on travis.

Issue reported by @cgohlke 

http://matplotlib.1069221.n5.nabble.com/Matplotlib-devel-First-release-candidate-for-the-1-5-0-series-tp46156p46166.html

http://stackoverflow.com/questions/28499580/matplotlib-import-error-backend-gdk-soundefined-symbol-pyarray-setbaseobjec

The code in question went in in 
https://github.com/matplotlib/matplotlib/pull/3665

which was in response to discussion at

http://thread.gmane.org/gmane.comp.python.matplotlib.general/34771/focus=34771

Options here are: 

 - fix code to address both issues (no idea how hard that would be)
 - up the minimal numpy until it works

I am in favor of bumping the numpy version, but do not feel strongly.

### PR DESCRIPTION
**Pull request by jkseppan**
Fixes #5078, at least according to a quick test with `import matplotlib.backends.backend_gdk`

### CODE REVIEW

**Comment by OceanWolf** [comment]
Hmm, trying to restart the 3.4 test but it doesn't want to... Travis says it restarted successfully, but it hasn't restarted...

File: src/_backend_gdk.c [hunk 1]
```diff
@@ -52,7 +52,15 @@ static PyObject *pixbuf_get_pixels_array(PyObject *self, PyObject *args)
...
     Py_INCREF(py_pixbuf);
-    PyArray_SetBaseObject(array, (PyObject *)py_pixbuf);
+#if NPY_API_VERSION >= 0x00000007
+    if (PyArray_SetBaseObject(array, (PyObject *)py_pixbuf) == -1) {
+        Py_DECREF(py_pixbuf);
+        Py_DECREF(array);
+        return NULL;
+    }
+#else
+    PyArray_BASE(array) = py_pixbuf;
```
Comment (WeatherGod) [inline src/_backend_gdk.c]: Do we need to cast the `py_pixbuf` pointer into a `PyObject` pointer? `py_pixbuf` is type `PyGObject*`.

Comment (mdehoon) [inline src/_backend_gdk.c]: Yes, because PyArray_SetBaseObject expects a PyObject pointer as its second argument.

Comment (jkseppan) [inline src/_backend_gdk.c]: > On 16 Sep 2015, at 17:18, mdehoon <notifications@github.com> wrote:
> 
> In src/_backend_gdk.c <https://github.com/matplotlib/matplotlib/pull/5080#discussion_r39634884>:
> 
> > @@ -52,7 +52,15 @@ static PyObject *pixbuf_get_pixels_array(PyObject *self, PyObject *args)
> >  
> >      /* the array holds a ref to the pixbuf pixels through this wrapper*/
> >      Py_INCREF(py_pixbuf);
> > -    PyArray_SetBaseObject(array, (PyObject *)py_pixbuf);
> > +#if NPY_API_VERSION >= 0x00000007
> > +    if (PyArray_SetBaseObject(array, (PyObject *)py_pixbuf) == -1) {
> > +        Py_DECREF(py_pixbuf);
> > +        Py_DECREF(array);
> > +        return NULL;
> > +    }
> > +#else
> > +    PyArray_BASE(array) = py_pixbuf;
> Yes, because PyArray_SetBaseObject expects a PyObject pointer as its second argument.
> 
I guess the question is whether we need to do the cast in the #else branch too. It’s not required in C, but assigning to a pointer to a different type can cause a compiler warning, so it would be slightly cleaner to cast in that branch too.

Comment (WeatherGod) [inline src/_backend_gdk.c]: Correct. That was the question I was asking.

**Comment by tacaswell** [comment]
All the tests pass and cleans up a corner-case failure mode.
